platform/src/pages/collocation/reports/monitor_report/[device].jsx

```diff
@@ -0,0 +1,94 @@
+import Layout from '@/components/Layout';
+import NavigationBreadCrumb from '@/components/Navigation/breadcrumb';
+import { useRouter } from 'next/router';
+import DataCompletenessTable from '@/components/Collocation/Report/MonitorReport/DataCompletenessTable';
+import { wrapper } from '@/lib/store';
+import {
+  useGetCollocationResultsQuery,
+  getRunningQueriesThunk,
+  getCollocationResults,
+} from '@/lib/store/services/collocation';
+import { useState } from 'react';
+import InterCorrelationChart from '@/components/Collocation/Report/MonitorReport/InterCorrelation';
+import IntraCorrelationChart from '@/components/Collocation/Report/MonitorReport/IntraCorrelation';
+import Toast from '@/components/Toast';
+
+export const getServerSideProps = wrapper.getServerSideProps((store) => async (context) => {
+  const name = context.params?.name;
+  if (typeof name === 'string') {
+    store.dispatch(getCollocationResults.initiate(name));
+  }
+
+  await Promise.all(store.dispatch(getRunningQueriesThunk()));
+
+  return {
+    props: {},
+  };
+});
+
+const MonitorReport = () => {
+  const router = useRouter();
+  const { device, startDate, endDate } = router.query;
+
+  const {
+    data: data,
+    isLoading,
+    isSuccess,
+    isError,
+    error,
+  } = useGetCollocationResultsQuery({
+    devices: device,
+    startDate: startDate,
+    endDate: endDate,
+  });
+  let collocationResults = data?.data;
+
+  const [correlationDevices, setCorrelationDevices] = useState([device]);
```

Comment:
The name of this file has special characters(`[` and `]`).
Could this have been an oversight? Note that it's not a best practice to have special characters in file names.

Reply:
It's a Nextjs feature known as dynamic routes. Explained in this easy to understand [documentation](https://nextjs.org/docs/routing/dynamic-routes)

Reply:
Oh, thanks. Didn't know they were dynamic routes